lib/Backend/NativeCodeGenerator.cpp

```diff
@@ -3268,24 +3268,34 @@ NativeCodeGenerator::QueueFreeNativeCodeGenAllocation(void* codeAddress, void *
     // OOP JIT will always queue a job
 
     // The foreground allocators may have been used
-    if(this->foregroundAllocators && this->foregroundAllocators->emitBufferManager.FreeAllocation(codeAddress))
+    if(this->foregroundAllocators)
     {
-        return;
+#if PDATA_ENABLED && defined(_WIN32)
+        if (functionTable)
+        {
+            NtdllLibrary::Instance->DeleteGrowableFunctionTable(functionTable);
```

Comment:
i think we can only do this on win8+?

Reply:
I think if it's not win8+ we are using pdata field so functionTable should be nullptr? maybe adding an assertion...

Reply:
Ah, that makes sense.